apps/core/components/bookingContext/BookingContext.js

```diff
@@ -0,0 +1,54 @@
+// @flow
+
+import * as React from 'react';
+import { withContext, noop } from '@kiwicom/margarita-utils';
+import { type PassengerCardType } from '@kiwicom/margarita-components';
+
+type Props = {|
+  +children: React.Node,
+|};
+
+type State = {|
+  +passengers: Array<PassengerCardType>,
+  +actions: {|
+    +setPassengers: (Array<PassengerCardType>) => void,
+  |},
+|};
+
+const defaultState = {
+  passengers: [],
+  actions: {
+    setPassengers: noop,
+  },
+};
+
+const { Provider, Consumer } = React.createContext<State>(defaultState);
+
+export default class BookingContextProvider extends React.Component<
+  Props,
+  State,
+> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      ...defaultState,
+      actions: {
+        setPassengers: this.setPassengers,
+      },
+    };
+  }
+
+  setPassengers = (passengers: Array<PassengerCardType>) => {
```

Comment:
Maby just the `PassengerType` will be better now? what do you think?

Reply:
You mean creating `type PassengerType = Array<PassengerCardType>`?

Reply:
No `PaseangerCardType` is too coupled with the `PassengerCard` component and now we want to use it in many places, and I think that type should be more related to Context than the Component because the state is the source of truth.

But it is just a proposal. To have the `passengerType` inside the context.

Reply:
Done! now `PassengerType` is in the context and is used by `PassengerCard`